src/main/ParseArgs.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import yargs from 'yargs';
+
+import {protocols} from '../../electron-builder.json';
+
+import * as Validator from './Validator';
+
+export default function parse(args) {
+  return validateArgs(parseArgs(triageArgs(args)));
+}
+
+function triageArgs(args) {
+  // ensure any args following a possible deeplink are discarded
+  if (protocols && protocols[0] && protocols[0].schemes && protocols[0].schemes[0]) {
+    const scheme = protocols[0].schemes[0];
+    const deeplinkIndex = args.findIndex((arg) => arg.includes(`${scheme}:`));
+    if (deeplinkIndex !== -1) {
+      return args.slice(0, deeplinkIndex + 1);
+    }
+  }
+  return args;
+}
+
+function parseArgs(args) {
+  return yargs.
+    boolean('squirrel-uninstall').
```

Comment:
I assume we also have to support these squirrel-specific args still?

Reply:
Alas, yes. Unless we remove squirrel altogether, but then we'll have to solve for no notifications etc.

No longer in master, but we still have to solve for notifications etc. there. 😛